### PR TITLE
DAOS-623 cq: Python/flake cleanups.

### PR DESCRIPTION
**Pull request by ashleypittman**
Enable GitHub actions checking of scons files with flake.
Turn on warnings for almost all remaining error classes.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/tests/ftest/pool/create_all_hw.py [hunk 1]
```diff
@@ -81,21 +80,17 @@ def test_one_pool(self):
...
-        self.assertLessEqual(
-                abs(self.smd_avail_bytes - tier_bytes[1]), self.delta_bytes,
-                "Invalid SMD size: want={}, got={}, delta={}".format(
-                    self.smd_avail_bytes, tier_bytes[1], self.delta_bytes))
+        self.assertLessEqual(abs(self.scm_avail_bytes - tier_bytes[0]), self.delta_bytes,
+                             "Invalid SCM size: want={}, got={}, delta={}".format(
+                             self.scm_avail_bytes, tier_bytes[0], self.delta_bytes))
+        self.assertLessEqual(abs(self.smd_avail_bytes - tier_bytes[1]), self.delta_bytes,
+                             "Invalid SMD size: want={}, got={}, delta={}".format(
+                             self.smd_avail_bytes, tier_bytes[1], self.delta_bytes))
```
Comment (daltonbohning) [inline src/tests/ftest/pool/create_all_hw.py:88]: Was the previous format invalid? These look like they were okay

File: src/tests/ftest/pool/create_all_hw.py [hunk 2]
```diff
@@ -244,37 +232,33 @@ def test_two_pools(self):
...
         self.scm_avail_bytes, self.smd_avail_bytes = self.get_pool_available_bytes()
 
         self.log.info("Checking size of available storage after the creation of the second pool")
         self.assertLessEqual(
-                self.scm_avail_bytes, self.delta_bytes,
-                "Invalid SCM size: want=0, got={}, delta={}".format(
-                    self.scm_avail_bytes, self.delta_bytes))
+            self.scm_avail_bytes, self.delta_bytes,
+            "Invalid SCM size: want=0, got={}, delta={}".format(
+                self.scm_avail_bytes, self.delta_bytes))
```
Comment (daltonbohning) [inline src/tests/ftest/pool/create_all_hw.py:260]: E.g. this format is similar to many that were changed above

**Review by daltonbohning — DISMISSED** [review]

File: src/tests/ftest/pool/create_all_hw.py (hunk 1 above)
Comment (ashleypittman) [reply src/tests/ftest/pool/create_all_hw.py:88]: It took me ages to get these to pass, yes they were incorrect.

File: src/tests/ftest/pool/create_all_hw.py (hunk 2 above)
Comment (ashleypittman) [reply src/tests/ftest/pool/create_all_hw.py:260]: I don't get it either, like you say they look very similar.

**Review by jolivier23 — DISMISSED** [review]

File: src/tests/ftest/pool/create_all_hw.py (hunk 1 above)
Comment (ashleypittman) [reply src/tests/ftest/pool/create_all_hw.py]: @phender  fixed this file in https://github.com/daos-stack/daos/pull/10718 so I've updated this PR and it no longer changes this file at all.

File: src/tests/ftest/util/fio_test_base.py [hunk 3]
```diff
@@ -1,4 +1,3 @@
-#!/usr/bin/python3
 """
   (C) Copyright 2020-2022 Intel Corporation.
```
Comment (ashleypittman) [inline src/tests/ftest/util/fio_test_base.py:2]: Copyright hook missed this file!

**Review by daltonbohning — DISMISSED** [review]

File: src/tests/ftest/util/ec_utils.py [hunk 4]
```diff
@@ -73,9 +73,7 @@ def check_aggregation_status(log, pool, quick_check=True, attempt=20):
 
 
 class ErasureCodeIor(ServerFillUp):
-    # pylint: disable=too-many-ancestors
-    """
-    Class to used for EC testing.
+    """Class to used for EC testing.
```
Comment (kccain) [inline src/tests/ftest/util/ec_utils.py:76]: extremely minor: is this meant to say "Class to be used for EC testing"

**Review by kccain — DISMISSED** [review]

**Pull request merged**